src/board/square.rs

```diff
@@ -0,0 +1,59 @@
+use super::{Board, errors::InvalidSquareError, Orientation};
+
+// Represents a square on the board
```

Comment:
Since you're handling and moving `Square` and adding new types, could you replace docs with `///` comments instead of `//` comments? That would support `rustdoc` iirc